### PR TITLE
MAINT: Post 0.11.1 release, drop py2, qt4

### PR DESCRIPTION
**Pull request by ksunden**
This focuses on distribution, tests, and docs. This may not be comprehensive, but covers the cases I found by looking and a few greps

Noteably, this does not change any actual internal code yet, to avoid merge conflicts with pending PRs.

### CODE REVIEW

File: doc/source/installation.rst [hunk 1]
```diff
@@ -3,8 +3,8 @@ Installation
 
 PyQtGraph depends on:
 
-* Python 2.7 or Python 3.x
-* A Qt library such as PyQt4, PyQt5, PySide, or PySide2
+* Python 3.7+
```
Comment (lidstrom83) [inline doc/source/installation.rst:6]: Should this be 3.6? Above it looks like we're keeping Python 3.6 for now.

File: README.md [hunk 2]
```diff
@@ -19,10 +19,23 @@ heavy leverage of numpy for number crunching, Qt's GraphicsView framework for
...
+
+* All minor versions of Python released 42 months prior to the project, and at minimum the two latest minor versions.
+
+* All minor versions of numpy released in the 24 months prior to the project, and at minimum the last three minor versions.
+* All minor versions of Qt 5 currently supported by upstream Qt (Note, Qt 6 support is not yet implemented)
+
+
+Currently this means:
+
+* Python 3.7+
```
Comment (lidstrom83) [inline README.md:34]: Here too. That or these are correct and the matrix needs to drop 3.6.

**Comment by j9ac9k** [comment]
Nep-29 says we should not support 3.6, but 3.7+ From this point..

I'm working on migrating our CI to GitHub actions in another PR.

**Comment by lidstrom83** [comment]
> Nep-29 says we should not support 3.6, but 3.7+ From this point.. I'm working on migrating our CI to GitHub actions in another PR.
>

I noticed that and also saw you were working on the CI migration after making my dumb comments that I deleted :)

**Comment by j9ac9k** [comment]
> > Nep-29 says we should not support 3.6, but 3.7+ From this point.. I'm working on migrating our CI to GitHub actions in another PR.
> 
> >
> 
> 
> 
> I noticed that and also saw you were working on the CI migration after making my dumb comments that I deleted :)
> 
> 

My PR isn't a sure thing yet, there are some configs that are still being problematic, it may take me a while to resolve.

File: tox.ini [hunk 3]
```diff
@@ -9,9 +9,6 @@ envlist =
     ; qt 5.9.7 (LTS)
     py36-{pyqt5,pyside2}_59_conda
```
Comment (j9ac9k) [inline tox.ini:10]: I am ok with testing PyQt5.9 in tox; but we should test it with python 3.7.  

I suggest removing all references to `py36` from here, and adding some `py39` references where appropriate.

**Comment by j9ac9k** [comment]
This is such a relief to merge this...

**Comment by j9ac9k** [comment]
thank you @ksunden for helping make this happen!

**Pull request merged**